lib/remoteResource.js

```diff
@@ -51,45 +55,78 @@ const createRemoteResources = async (razeeApi, apiKey, subscriptions, clusterId)
         }
       };
       resourceTemplate.spec.requests.push(parsed);
+      
+      const secretJson = {
+        'apiVersion': 'v1',
+        'kind': 'Secret',
+        'metadata': {
+          'namespace': NAMESPACE,
+          'name': secretName,
+          'annotations': {
+            'deploy.razee.io/clustersubscription': sub.subscriptionUuid,
+            'deploy.razee.io/clusterid': clusterId
+          },
+          'labels': {
+            'razee/watch-resource': 'lite'
+          }
+        },
+        'data': {
+          'razee-api-org-key': apiKey
+        }
+      };
 
-      const opt = { simple: false, resolveWithFullResponse: true };
 
-      const uri = krm.uri({ name: resourceName, namespace: NAMESPACE });
-      log.debug(resourceName);
-      const get = await krm.get(resourceName, NAMESPACE, opt);
-      if (get.statusCode === 200) {
-        // the remote resource already exists so use mergePatch to apply the resource
-        log.info(`Attempting mergePatch for an existing resource ${uri}`);
-        const mergeResult = await krm.mergePatch(resourceName, NAMESPACE, resourceTemplate, opt);
-        if (mergeResult.statusCode === 200) {
-          log.info('mergePatch successful', { 'statusCode': mergeResult.statusCode, 'statusMessage': mergeResult.statusMessage });
-        } else {
-          log.error('mergePatch error', { 'statusCode': mergeResult.statusCode, 'statusMessage': mergeResult.statusMessage });
-        }
-      } else if (get.statusCode === 404) {
-        // the remote resource does not exist so use post to apply the resource
-        log.info(`Attempting post for a new resource ${uri}`);
-        const postResult = await krm.post(resourceTemplate, opt);
-        if (postResult.statusCode === 200 || postResult.statusCode === 201) {
-          log.info('post successful', { 'statusCode': postResult.statusCode, 'statusMessage': postResult.statusMessage });
-        } else {
-          log.error('post error', { 'statusCode': postResult.statusCode, 'statusMessage': postResult.statusMessage });
-        }
-      } else {
-        log.error(`Get ${get.statusCode} ${uri}`);
-      }
+      await applyResource(RR_API_VERSION, 'RemoteResource', remoteResourceName, remoteResourceJson, NAMESPACE);
+      await applyResource('v1', 'Secret', secretName, secretJson, NAMESPACE);
```

Comment:
lets pull the necessary fields from the json itself, instead of passing everything in.
```suggestion
      await applyResource(RR_API_VERSION, 'RemoteResource', remoteResourceJson);
      await applyResource('v1', 'Secret', secretJson);
```